exploration-module/src/main/java/io/edpn/backend/exploration/adapter/config/ServiceConfig.java

```diff
@@ -39,12 +44,15 @@ public ReceiveNavRouteService receiveNavRouteService(
             LoadSystemCoordinateRequestBySystemNamePort loadSystemCoordinateRequestBySystemNamePort,
             DeleteSystemCoordinateRequestPort deleteSystemCoordinateRequestPort,
             KafkaSystemCoordinatesResponseMapper kafkaSystemCoordinatesResponseMapper,
+            LoadSystemEliteIdRequestBySystemNamePort loadSystemEliteIdRequestBySystemNamePort,
+            DeleteSystemEliteIdRequestPort deleteSystemEliteIdRequestPort,
+            KafkaSystemEliteIdResponseMapper kafkaSystemEliteIdResponseMapper,
             MessageMapper messageMapper,
             @Qualifier("explorationObjectMapper") ObjectMapper objectMapper,
             @Qualifier("explorationRetryTemplate") RetryTemplate retryTemplate,
             @Qualifier("explorationThreadPoolTaskExecutor") Executor executor
     ) {
-        return new ReceiveNavRouteService(createSystemPort, loadSystemPort, saveSystemPort, sendMessagePort, loadSystemCoordinateRequestBySystemNamePort, deleteSystemCoordinateRequestPort, kafkaSystemCoordinatesResponseMapper, messageMapper, objectMapper, retryTemplate, executor);
+        return new ReceiveNavRouteService(createSystemPort, loadSystemPort, saveSystemPort, sendMessagePort, loadSystemCoordinateRequestBySystemNamePort, deleteSystemCoordinateRequestPort, kafkaSystemCoordinatesResponseMapper, loadSystemEliteIdRequestBySystemNamePort, deleteSystemEliteIdRequestPort, kafkaSystemEliteIdResponseMapper, messageMapper, objectMapper, retryTemplate, executor);
```

Comment:
Wasn't there some advice once upon a time about the maximum number of parameters a method/constructor should take, before thinking about refactoring? 😄

Reply:
And on a more serious note, maybe vertical formatting for these, for readability?

Reply:
the point IS to have all the ports independently. The only place where you could reduce the amount of parameters is at the bean creation input. which I think is even worse style, and not that it fixes a lot.


```suggestion
    @Bean(name = "explorationReceiveNavRouteService")
    public ReceiveNavRouteService receiveNavRouteService(
            SystemRepository systemRepository,
            SendMessagePort sendMessagePort,
            SystemCoordinateRequestRepository systemCoordinateRequestRepository,
            KafkaSystemCoordinatesResponseMapper kafkaSystemCoordinatesResponseMapper,
            SystemEliteIdRequestRepository systemEliteIdRequestRepository,
            KafkaSystemEliteIdResponseMapper kafkaSystemEliteIdResponseMapper,
            MessageMapper messageMapper,
            @Qualifier("explorationObjectMapper") ObjectMapper objectMapper,
            @Qualifier("explorationRetryTemplate") RetryTemplate retryTemplate,
            @Qualifier("explorationThreadPoolTaskExecutor") Executor executor
    ) {
        return new ReceiveNavRouteService(
                systemRepository,
                systemRepository,
                systemRepository,
                sendMessagePort,
                systemCoordinateRequestRepository,
                systemCoordinateRequestRepository,
                kafkaSystemCoordinatesResponseMapper,
                systemEliteIdRequestRepository,
                systemEliteIdRequestRepository,
                kafkaSystemEliteIdResponseMapper,
                messageMapper,
                objectMapper,
                retryTemplate,
                executor);
    }
```

this only happens because we are managing our own beans creation. If we were to use the spring automagic glue this would all be obfuscated.